### PR TITLE
Feature/update models

### PR DESCRIPTION
**Pull request by TheTechArch**
Update resource registry

### CODE REVIEW

File: src/Altinn.ResourceRegistry.Persistence/Migration/v0.10-resource-registry/02-setup-procedures.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+-- Procedure: search_for_resource
```
Comment (jonkjetiloye) [inline src/Altinn.ResourceRegistry.Persistence/Migration/v0.10-resource-registry/02-setup-procedures.sql:1]: We've been doing 0.01 increments on the migration script version in other repos, but I'm a bit unsure how the version number actually affects the execution order of the scripts (if anyone makes a v0.0x script in the future)

File: src/Altinn.ResourceRegistry.Core/Models/ServiceResource.cs [hunk 2]
```diff
@@ -32,32 +32,33 @@ public class ServiceResource
...
         /// </summary>
-        public DateTime ValidFrom { get; set; } 
+        public List<string>? Spatial { get; set;  }
 
         /// <summary>
-        /// When the resource is available to
+        /// List of possible contact points
         /// </summary>
-        public DateTime ValidTo { get; set; }
+        public List<string> ContactPoints { get; set; }
```
Comment (jonkjetiloye) [inline src/Altinn.ResourceRegistry.Core/Models/ServiceResource.cs:51]: Don't ContactPoint need to be created as a more complex model to indicate the type of contact point etc?

File: test/ResourceRegistryTest/ResourceControllerTest.cs [hunk 3]
```diff
@@ -771,13 +764,17 @@ public async Task UpdateResource_Ok()
             ServiceResource resource = new ServiceResource()
             {
                 Identifier = "altinn_access_management",
+                Title = new Dictionary<string, string>() { { "en", "Maskinporten Scope Delegation" } },
+                Description = new Dictionary<string, string>() { { "nb", "Maskinporten Scope Delegation" } },
+                RightDescription = new Dictionary<string, string> { { "nb", "Access to Maskinporten Scope Delegation" } },
+                Status = "Completed",
```
Comment (jonkjetiloye) [inline test/ResourceRegistryTest/ResourceControllerTest.cs:770]: This resource shouldn't be complete if we now are to require that text resources are provided in all three languages?

Also: We also decided MainLanguage was no longer needed right, if we require all languages?

File: src/Altinn.ResourceRegistry.Core/Models/ServiceResource.cs (hunk 2 above)
Comment (TheTechArch) [reply src/Altinn.ResourceRegistry.Core/Models/ServiceResource.cs]: Changed to more comples

File: test/ResourceRegistryTest/ResourceControllerTest.cs (hunk 3 above)
Comment (TheTechArch) [reply test/ResourceRegistryTest/ResourceControllerTest.cs:770]: For me the status has nothing to do with the completness of resource registry resource
It is only related to the status of the service itself

Removed main langauge

File: src/Altinn.ResourceRegistry.Core/Enums/ResourcePartyType.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Altinn.ResourceRegistry.Core.Enums
+{
+    /// <summary>
+    /// Defines the type of party that a resource is targeting
+    /// </summary>
+    public enum ResourcePartyType
+    {
+        [EnumMember(Value = "PrivatePerson")]
+        PrivatePerson = 0,
```
Comment (jonkjetiloye) [inline src/Altinn.ResourceRegistry.Core/Enums/ResourcePartyType.cs:11]: Burde denne hatt en default verdi?

File: src/Altinn.ResourceRegistry.Core/Models/ServiceResource.cs [hunk 5]
```diff
@@ -106,9 +112,9 @@ public class ServiceResource
         public ResourceType ResourceType { get; set; }
 
         /// <summary>
-        /// The fallback language of the resource
+        /// Available for type defines which type of entity / person that resource targets
         /// </summary>
-        public string MainLanguage { get; set; } = "nb";
+        public List<ResourcePartyType>? AvailableForType { get; set; }
```
Comment (jonkjetiloye) [inline src/Altinn.ResourceRegistry.Core/Models/ServiceResource.cs:117]: Also missing the Delegable and Visible bool flags

Comment (jonkjetiloye) [reply src/Altinn.ResourceRegistry.Core/Models/ServiceResource.cs:117]: My bad github hiding lines from me

**Pull request merged**